keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,931 @@
 
-The delete workflow is described as a synchronous, but it will likely be asynchronous to accommodate potentially long delete times when buckets contain many objects. The steps should still mostly follow what's outlined below.
+A BucketAccess with condition `AccessGranted` set to `True`, and pointing to a secret by the name specified in the BucketAccessRequest is required for the Pod to be started. If either the Bucket hasn't been provisioned, or the access has not been granted - the pod will not run. It will wait indefinitely for those two conditions to be true.
 
-Here is the workflow:
+In order to specify the bucket for the pod, a projected volume pointing to the secret mentioned above needs to be specified.
 
-+ User deletes their `BucketRequest` (BR) which is deferred until finalizers have been removed.
-+ COSI gets the `Bucket` instance from the BR and notes the retention policy.
-+ If the `Bucket` instance was created by COSI then COSI deletes the `Bucket`, which sets its deleteTimestamp but does not delete it due to finalizer.
-+ If the `Bucket` instance was not created by COSI then the `Bucket` is not deleted, but finalizers are removed so that the admin can manually delete it.
-+ COSI unbinds the BR from the `Bucket`.
-+ Sidecar sees the `Bucket` is unbound, and if the retention policy is "Delete", the gRPC calls the provisoner's _Delete_ interface. Upon successful completion, the `Bucket` is updated to Deleted.
-+ If the retention policy is "Retain" then the `Bucket` remains "Released" and it can potentially be reused.
-+ When COSI sees the `Bucket` is "Deleted", it removes all finalizers and the resources are garbage collected.
+```
+    PodSpec - pod1
+    |-------------------------------------------------|
+    | spec:                                           |
+    |   containers:                                   |
+    |   - volumeMounts:                               |
+    |       name: cosi-bucket                         |
+    |       mountPath: /cosi/bucket1                  |
+    | volumes:                                        |
+    | - name: cosi-bucket                             |
+    |   projected:                                    |
+    |     sources:                                    |
+    |     - secret:                                   |
+    |       name: bucketcreds1                        |
+    |-------------------------------------------------|
+```
 
-## Grant Bucket Access
-This workflow describes the automation supporting granting access to an existing backend bucket.
+The volume `mountPath` will be the directory where bucket credentials and other related information will be served.
 
-Here is the workflow:
+NOTE: the contents of the files served in mountPath will be a COSI generated file containing credentials and other information required for accessing the bucket. This is not intended to specify a mountpoint containing the bucket's data.
 
```

Comment:
what all information will be available in this file, is it include endpoint + credentials? I suppose the COSI- the controller will be the one creating this secret right, does it have a specific way to generate the secret. IMO it will be better if there is the chance to split into different files than in a single. Can you please share the how this file may look like?